sdk/jobs/automl-standalone-jobs/automl-image-classification-multiclass-batch-scoring/image-classification-multiclass-batch-scoring-non-mlflow-model.ipynb

```diff
@@ -0,0 +1,842 @@
+    "\n",
+    "**Please Note** that The name of the endpoint must be unique in the Azure region. For more information on the naming rules, see [managed endpoint limits](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-quotas#azure-machine-learning-managed-online-endpoints). \n",
+    "\n",
+    "Optionally, you can add description, tags to your endpoint."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.ai.ml.entities import (\n",
+    "    BatchEndpoint,\n",
+    "    BatchDeployment,\n",
+    "    BatchRetrySettings,\n",
+    ")\n",
+    "from azure.ai.ml.constants import BatchDeploymentOutputAction\n",
+    "\n",
+    "import datetime\n",
+    "\n",
+    "batch_endpoint_name = \"my-batch-endpoint-\" + datetime.datetime.now().strftime(\n",
+    "    \"%Y%m%d%H%M\"\n",
+    ")\n",
+    "\n",
+    "# create a batch endpoint\n",
+    "endpoint = BatchEndpoint(\n",
+    "    name=batch_endpoint_name,\n",
+    "    description=\"this is a sample batch endpoint\",\n",
+    "    tags={\"foo\": \"bar\"},\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Using the MLClient created earlier, we'll now create the Endpoint in the workspace. This command will start the endpoint creation and return a confirmation response while the endpoint creation continues."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ml_client.begin_create_or_update(endpoint)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### 5.4.2 Configure the deployment \n",
+    "\n",
+    "A deployment is a set of resources required for hosting the model that does the actual inferencing. We'll create a deployment for our endpoint using the BatchDeployment class.\n",
+    "\n",
+    "Few important parameters are\n",
+    "\n",
+    "- **model**: Link to the registered model\n",
+    "\n",
+    "- **code_path**: Folder containing the scoring script\n",
+    "\n",
+    "- **scoring_script**: Path of the scoring script relative to the `code path` parameter.\n",
+    "\n",
+    "- **environment**: Environment object which would be used to create the virtual environment for inferencing. It should of type `azure.ai.ml.entities.environment` "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# create a batch deployment\n",
+    "deployment = BatchDeployment(\n",
+    "    name=\"non-mlflow-deployment\",\n",
+    "    description=\"this is a sample non-mlflow deployment\",\n",
+    "    endpoint_name=batch_endpoint_name,\n",
+    "    model=registered_model.id,\n",
+    "    code_path=\"./scripts\",\n",
+    "    scoring_script=\"batch_scoring.py\",\n",
+    "    environment=env,\n",
+    "    compute=\"cpu-cluster\",\n",
```

Comment:
Sure. I added a step to create a new computer target for deployment with [1,5] number of nodes in the cluster, so that the custom know how many machines would be used by the mangedEndpoint at the peek of the traffic. Would that be fine?

Reply:
Sure Nilesh. Please use compute creation code similar to how we have in the steps above. Some like this.

from azure.ai.ml.entities import AmlCompute
from azure.core.exceptions import ResourceNotFoundError

compute_name = "cpu-cluster"

try:
    _ = ml_client.compute.get(compute_name)
    print("Found existing compute target.")
except ResourceNotFoundError:
    print("Creating a new compute target...")
    compute_config = AmlCompute(
        name=compute_name,
        type="amlcompute",
        size="Standard_DS3_V2",
        idle_time_before_scale_down=120,
        min_instances=0,
        max_instances=5,
    )
    ml_client.begin_create_or_update(compute_config)